idb/postgres/internal/writer/writer_test.go

```diff
@@ -354,38 +383,24 @@ func TestWriterTxnParticipationTableBasic(t *testing.T) {
 	err = pgutil.TxWithRetry(db, serializable, f, nil)
 	require.NoError(t, err)
 
-	rows, err := db.Query(
-		context.Background(), "SELECT * FROM txn_participation ORDER BY round, intra, addr")
-	require.NoError(t, err)
-	defer rows.Close()
+	results, err := txnParticipationQuery(db, `SELECT * FROM txn_participation ORDER BY round, intra, addr`)
+	assert.NoError(t, err)
+	assert.Len(t, results, 3)
 
-	var addr []byte
-	var round uint64
-	var intra uint64
+	// check address
```

Comment:
I think it would be better to do `assert.Equal(t, txnParticipationRow{...}, results[0])`.